cli/pack/common.go

```diff
@@ -458,9 +458,14 @@ func prepareDefaultPackagePaths(opts *config.CliOpts, packagePath string) {
 func getVersion(packCtx *PackCtx, opts *config.CliOpts, defaultVersion string) string {
 	packageVersion := defaultVersion
 	if packCtx.Version == "" {
-		// Get version from git only if packing an application from the current directory.
-		if opts.App.InstancesEnabled == "." {
-			version, err := util.CheckVersionFromGit(opts.App.InstancesEnabled)
+		// Get version from git only if packing an application from the current directory,
+		// or packing with cartridge-compat enabled.
+		var appPath = opts.App.InstancesEnabled
+		if packCtx.CartridgeCompat {
+			appPath = filepath.Join(appPath, packCtx.Name)
```

Comment:
If instances enabled == "." what will be in appPath? ./app_name? How does this work?

Reply:
If `cartridge-compat` will be enabled, appPath will be equal to `instances_enabled_dir/app_name`.
Otherwise, it will be equal to `instances_enabled_dir`.

Reply:
> If cartridge-compat will be enabled, appPath will be equal to `instances_enabled_dir/app_name`.

Note that it will be true if `instances_enabled_dir` is not a `.`.

Reply:
> Note that it will be true if `instances_enabled_dir` is not a `.`. Otherwise, appName will be equal to `.`.
Now - yes :)